docs/Concepts/Events-and-Logs.md

```diff
@@ -2,21 +2,24 @@
 
 When a transaction is mined, smart contracts emit events and write logs to the blockchain.  
 
-Logs are associated with the contract address and included in the blockchain but logs are not accessible 
-from within contracts. Log storage is cheaper than contract storage (that is, it costs less gas) so if the required data can 
-be stored in and accessed from logs, the cost is reduced. For example, you can use logs to display all 
-transfers made using a specific contract but not the current state of the contract. 
+Logs are associated with the contract address and included in the blockchain but logs are not accessible
+from within contracts. Log storage is cheaper than contract storage (that is, it costs less gas) so if the required data can
+be stored in and accessed from logs, the cost is reduced. For example, you can use logs to display all
+transfers made using a specific contract but not the current state of the contract.
 
-A Dapp front end can either request logs using the [JSON-RPC API filter methods](../HowTo/Interact/Filters/Accessing-Logs-Using-JSON-RPC.md) 
-or subscribe to logs using the [RPC Pub/Sub API](../HowTo/Interact/APIs/RPC-PubSub.md#logs). 
+A Dapp front end can either request logs using the [JSON-RPC API filter methods](../HowTo/Interact/Filters/Accessing-Logs-Using-JSON-RPC.md)
+or subscribe to logs using the [RPC Pub/Sub API](../HowTo/Interact/APIs/RPC-PubSub.md#logs).
 
-## Topics 
+Improve log retrieval performance by using [`admin_generateLogBloomCache`](../Reference/API-Methods.md#admin_generatelogbloomcache) to
```

Comment:
Suggest rewording to make active: 

Use [`admin_generateLogBloomCache`](../Reference/API-Methods.md#admin_generatelogbloomcache) to improve log retrieval performance.